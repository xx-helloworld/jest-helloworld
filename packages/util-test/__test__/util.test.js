import { createVue } from "@xx-helloworld/jest-util";
import Child from "../child.vue";
import Parent from "../parent.vue";
import Vue from "vue";

describe("Inject", () => {
  it("test util", () => {
    const msg = "new message";
    const childMsg = "new child message";

    Vue.component("child", Child);
    Vue.component("parent", Parent);

    // const parent = {
    //   render(h) {
    //     return h(Parent, { props: { msg } }, [
    //       h(Child, { props: { childMsg } })
    //     ]);
    //   }
    // };

    const vm = createVue(
      {
        template: `
        <parent msg="msg">
          <child childMsg="childMsg" />
        </parent>
      `,
        data() {
          return {
            childMsg,
            msg
          };
        }
      },
      true,
      Vue
    );

    console.log(vm.$el.outerHTML);
    expect(vm.$el.outerHTML).toMatchSnapshot();
  });
});
