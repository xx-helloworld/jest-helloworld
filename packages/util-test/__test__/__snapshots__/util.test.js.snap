// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inject test util 1`] = `"<div><span>msg</span> <h1>childMsg</h1></div>"`;
